homeassistant/components/google_assistant/smart_home.py

```diff
@@ -107,6 +110,9 @@ def sync_serialize(self):
         if not traits:
             return None
 
+        device_type = (entity_config.get(CONF_TYPE) or
```

Comment:
We should not offer this functionality. We should just do the right thing. Please remove this.

Reply:
We should never put the burden on the user. Instead, a new air purifier device class should be added to the fan component. (here is [an example](https://developers.home-assistant.io/docs/en/entity_binary_sensor.html#available-device-classes) how binary sensor does it)

Reply:
How would you define the right thing, when the google assistant have more device types that HASS? For example, [Google Device Types](https://developers.google.com/actions/smarthome/guides/) define `outlet` and `switch`, while HASS have only switch. The same is with `fan` and `airfilter`.  

For me ability to choose the type is the same type of configuration as ability to overwrite the name of device for google.

Reply:
Followup MR from this discussion: https://github.com/home-assistant/home-assistant/pull/17866